daemon/container.go

```diff
@@ -559,19 +532,69 @@ func (container *Container) buildHostnameFile() error {
 	return ioutil.WriteFile(container.HostnamePath, []byte(container.Config.Hostname+"\n"), 0644)
 }
 
-func (container *Container) buildHostsFiles(IP string) error {
+func (container *Container) buildJoinOptions() ([]libnetwork.EndpointOption, error) {
+	var (
+		joinOptions []libnetwork.EndpointOption
+		err         error
+		dns         []string
+		dnsSearch   []string
+	)
+
+	joinOptions = append(joinOptions, libnetwork.JoinOptionHostname(container.Config.Hostname),
+		libnetwork.JoinOptionDomainname(container.Config.Domainname))
+
+	if container.hostConfig.NetworkMode.IsHost() {
+		joinOptions = append(joinOptions, libnetwork.JoinOptionUseDefaultSandbox())
+	}
 
-	hostsPath, err := container.GetRootResourcePath("hosts")
+	container.HostsPath, err = container.GetRootResourcePath("hosts")
 	if err != nil {
-		return err
+		return nil, err
+	}
+	joinOptions = append(joinOptions, libnetwork.JoinOptionHostsPath(container.HostsPath))
+
+	container.ResolvConfPath, err = container.GetRootResourcePath("resolv.conf")
+	if err != nil {
+		return nil, err
 	}
-	container.HostsPath = hostsPath
+	joinOptions = append(joinOptions, libnetwork.JoinOptionResolvConfPath(container.ResolvConfPath))
 
-	var extraContent []etchosts.Record
+	if len(container.hostConfig.Dns) > 0 {
+		dns = container.hostConfig.Dns
```

Comment:
`docker/docker` code has a lot of these occurrences(infact this was not introduced by this PR) and  I don't think we can clean up all of that in this PR. So what I would suggest is that we should open a seperate PR and fix all of these and enable `golint` hook so that it catches these automatically. In libnetwork we do exactly that and we don't have any initialism violations.